src/hotspot/share/logging/logFileOutput.cpp

```diff
@@ -304,12 +300,33 @@ int LogFileOutput::write(const LogDecorations& decorations, const char* msg) {
   return written;
 }
 
+int LogFileOutput::write(const LogDecorations& decorations, const char* msg) {
+  if (_stream == NULL) {
+    // An error has occurred with this output, avoid writing to it.
+    return 0;
+  }
+
+  AsyncLogWriter* aio_writer = AsyncLogWriter::instance();
+  if (aio_writer != nullptr) {
+    aio_writer->enqueue(*this, decorations, msg);
+    return -1;
```

Comment:
This is interesting! Thanks for the comment. 
I take a look at logFileStreamOuptut.cpp again, `LogFileStreamOutput::write` returns -1 when it runs into some errors. 

I feel it's more appropriate to return 0 here.  The value suggests that 'succeed but the written bytes is unknown.'
so far,  it doesn't matter because nobody reads this return value.

Reply:
I see; `0` seems better then. Thanks for the clarification.